File: test-api.py
import requests
import json

print('\n🌐 Testing API Endpoints...\n')

# Use 127.0.0.1 explicitly for IPv4
BASE_URL = 'http://127.0.0.1:5000'

try:
    # Test 1: Register
    print('1️⃣ Testing Registration API...')
    response = requests.post(f'{BASE_URL}/api/register', json={
        'fullName': 'John Scientist',
        'email': 'john@example.com',
        'password': 'secure123'
    })
    print(f'   Status: {response.status_code}')
    print(f'   Response: {response.json()}')
    print()

    # Test 2: Login
    print('2️⃣ Testing Login API...')
    response = requests.post(f'{BASE_URL}/api/login', json={
        'email': 'john@example.com',
        'password': 'secure123'
    })
    print(f'   Status: {response.status_code}')
    data = response.json()
    print(f'   User: {data.get("user", {}).get("email")} - {data.get("user", {}).get("fullName")}')
    print()

    # Test 3: Get Profile
    print('3️⃣ Testing User Profile API...')
    response = requests.get(f'{BASE_URL}/api/user/john@example.com')
    print(f'   Status: {response.status_code}')
    data = response.json()
    print(f'   User: {data.get("email")} - {data.get("fullName")}')
    print()

    # Test 4: Save Structure
    print('4️⃣ Testing Save Structure API...')
    response = requests.post(f'{BASE_URL}/api/user/john@example.com/save-structure', json={
        'name': 'Water Molecule',
        'nodes': [
            {'id': 1, 'name': 'O', 'x': 100, 'y': 100},
            {'id': 2, 'name': 'H', 'x': 150, 'y': 80},
            {'id': 3, 'name': 'H', 'x': 150, 'y': 120}
        ],
        'bonds': [
            {'from': 1, 'to': 2, 'type': 'single'},
            {'from': 1, 'to': 3, 'type': 'single'}
        ]
    })
    print(f'   Status: {response.status_code}')
    data = response.json()
    print(f'   Structure: {data.get("structure", {}).get("name")} (ID: {data.get("structure", {}).get("id")})')
    print()

    # Test 5: Get Structures
    print('5️⃣ Testing Get Structures API...')
    response = requests.get(f'{BASE_URL}/api/user/john@example.com/structures')
    print(f'   Status: {response.status_code}')
    data = response.json()
    structures = data.get('structures', [])
    print(f'   Total Structures: {len(structures)}')
    for i, s in enumerate(structures):
        print(f'   [{i+1}] {s.get("name")} ({len(s.get("data", {}).get("nodes", []))} nodes)')
    print()

    print('✅ All API tests passed!\n')

except Exception as e:
    print(f'❌ Error: {e}\n')

File: test-api.js
// API Endpoint Tests
const http = require('http');

function makeRequest(method, path, data) {
    return new Promise((resolve, reject) => {
        const options = {
            hostname: 'localhost',
            port: 5000,
            path: path,
            method: method,
            headers: {
                'Content-Type': 'application/json'
            }
        };

        const req = http.request(options, (res) => {
            let body = '';
            res.on('data', chunk => body += chunk);
            res.on('end', () => {
                try {
                    resolve({
                        status: res.statusCode,
                        data: JSON.parse(body)
                    });
                } catch (e) {
                    resolve({
                        status: res.statusCode,
                        data: body
                    });
                }
            });
        });

        req.on('error', (err) => {
            console.error('Request error:', err.message);
            reject(err);
        });
        
        if (data) req.write(JSON.stringify(data));
        req.end();
    });
}

async function runTests() {
    console.log('\n🌐 Testing API Endpoints...\n');

    try {
        // Wait for server to be ready
        await new Promise(resolve => setTimeout(resolve, 1000));

        // Test 1: Register new user
        console.log('1️⃣ Testing Registration API...');
        try {
            const registerRes = await makeRequest('POST', '/api/register', {
                fullName: 'John Scientist',
                email: 'john@example.com',
                password: 'secure123'
            });
            console.log('   Status:', registerRes.status);
            console.log('   Response:', JSON.stringify(registerRes.data, null, 2).substring(0, 200));
            console.log();
        } catch (e) {
            console.log('   Error:', e.message);
            console.log();
        }

        // Test 2: Login
        console.log('2️⃣ Testing Login API...');
        try {
            const loginRes = await makeRequest('POST', '/api/login', {
                email: 'john@example.com',
                password: 'secure123'
            });
            console.log('   Status:', loginRes.status);
            console.log('   User:', loginRes.data.user?.email);
            console.log();
        } catch (e) {
            console.log('   Error:', e.message);
            console.log();
        }

        console.log('✅ API tests completed!\n');
        process.exit(0);
    } catch (error) {
        console.error('❌ Test failed:', error.message);
        process.exit(1);
    }
}

runTests();
